File: src/1281.subtract-the-product-and-sum-of-digits-of-an-integer.1406/solution.js
/**
 * @param {number} n
 * @return {number}
 */
var subtractProductAndSum = function (n) {
    let sum = 0;
    let product = 1;
    while (n) {
        const digit = n % 10;
        sum += digit;
        product *= digit;
        n = (n - digit) / 10;
    }
    return product - sum;
};

File: src/1281.subtract-the-product-and-sum-of-digits-of-an-integer.1406/solution.java
class Solution {
    public int subtractProductAndSum(int n) {
        int multi = 1;
        int sum = 0;
        while (n > 0) {
            int digit = n % 10;
            multi *= digit;
            sum += digit;
            n /= 10;
        }
        return multi - sum;
    }
}